frontend/src/components/form/ChangePasswordForm.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Heading,
+  Input,
+  InputGroup,
+  InputRightElement,
+  useBoolean,
+  VStack,
+} from "@chakra-ui/react";
+import { ViewIcon, ViewOffIcon } from "@chakra-ui/icons";
+import {
+  FieldValues,
+  SubmitErrorHandler,
+  SubmitHandler,
+  useForm,
+} from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import axios from "../../axios";
+import useFixedToast from "../../utils/hooks/useFixedToast";
+import {
+  ChangePasswordRequest,
+  ChangePasswordResponse,
+} from "../../proto/user-service";
+import { reset } from "../../feature/matching/matchingSlice";
+import { logout } from "../../feature/user/userSlice";
+
+function ChangePasswordForm() {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const toast = useFixedToast();
+  const [showPassword, setShowPassword] = useBoolean();
+
+  const validFormHandler: SubmitHandler<FieldValues> = (data) => {
+    const { password: newPassword } = data;
+
+    const changePasswordRequest: ChangePasswordRequest = {
+      newPassword,
+    };
+
+    axios
+      .post<ChangePasswordResponse>(
+        "/api/user/password",
+        changePasswordRequest,
+        {
+          withCredentials: true,
+        }
+      )
+      .then((res) => {
+        const { errorCode, errorMessage } = res.data;
+
+        if (errorCode) {
+          throw new Error(errorMessage);
+        }
+
+        toast.sendSuccessMessage(
+          "Your password is changed! You will need to login again!"
+        );
+
+        dispatch(reset());
+        dispatch(logout());
+        navigate("/login", { replace: true });
+      })
+      .catch((err) => {
+        toast.sendErrorMessage(err.message);
+      });
+  };
+
+  const invalidFormHandler: SubmitErrorHandler<FieldValues> = () => {
+    toast.sendErrorMessage(
+      "Please check if you have filled everything in correctly before submitting",
+      {
+        title: "Oops!",
+      }
+    );
+  };
+
+  return (
+    <VStack alignItems="flex-start">
+      <Heading as="h5" size="sm">
+        Change Password
+      </Heading>
+      <form onSubmit={handleSubmit(validFormHandler, invalidFormHandler)}>
+        <FormControl id="password" isInvalid={!!errors.password}>
+          <FormLabel>New Password</FormLabel>
+          <InputGroup>
```

Comment:
```suggestion
          <InputGroup>
          <PasswordInput register={register} formKey="password" />
```
1. This entire section has been abstracted to `<PasswordInput />` 
2. I think we should have a password confirmation too